test/e2e/broker_default_test.go

```diff
@@ -43,135 +42,229 @@ const (
 	eventType2        = "type2"
 	eventSource1      = "source1"
 	eventSource2      = "source2"
+	nilString         = "nilstring"
```

Comment:
I use this special string "nilstring" to indicate there is no extension. I'm not sure whether it's a good way to do it.

Reply:
with the map[string]string you can check the lenght, and figure out if there are no extensions.